### DISCUSSION TITLE
Can't get Lanraragi to open.

### DISCUSSION THREAD

**Discussion by Deejay85** [Difegue/LANraragi#964] [Q&A] [answered]
Several days ago Lanraragi just stopped opening for no apparent reason. I don't think it's with WSL, and given the error messages, I think it may be with the program itself.

Trying again in 2 seconds...
Redis error encountered: [KEYS] LOADING Redis is loading the dataset in memory,  at /usr/local/share/perl5/site_perl/Redis.pm line 321.

Trying again in 2 seconds...
Redis error encountered: Could not connect to Redis server at 127.0.0.1:6379: Connection refused at /home/koyomi/lanraragi/script/../lib/LANraragi/Model/Config.pm line 74.
        ...propagated at /usr/local/share/perl5/site_perl/Redis.pm line 673.

All it does it loop through these messages, and no matter how many times I restart the program, and run it fresh, it can't pull up Lanraragi interface when I try to pull up http://localhost:3000

**Comment by Difegue** [comment]
Tried a restart? WSL Redis is iffy sometimes.

**Reply by Deejay85** [reply]
Sure did. Also uninstalled/reinstalled WSL while I was at it...didn't make a bit of difference.

**Reply by Deejay85** [reply]
Still having the same problem over a week later...I got no idea how to fix this. Is there any other information that I need to provide to help fix it? Going from WSL 2 to WSL 1 didn't make a difference.

**Reply by Difegue** [reply, marked as answer]
I don't really have an answer here; LOADING errors are supposed to be temporary which is why we wait on them.
 Maybe the database itself is broken? You could check if simply removing the database rdb and aof files is enough to get things running again.

**Reply by Difegue** [reply]
Might be worth digging into the redis logfile in the app folder as well.

**Reply by Deejay85** [reply]
Somehow deleting the files you mentioned worked...I'm impressed. Also check the log file, and something caught my attention:

Bad file format reading the append only file appendonly.aof.63.incr.aof: make a backup of your AOF file, then use ./redis-check-aof --fix <filename.manifest>

That's interesting in that it highlighted the offending file only one question...how do I fix it? I assume it means to use a program other than Lanraragi to do that...right? I deleted it of course, which I assume is how it got to work again, but this is more for future reference.

**Reply by Difegue** [reply]
Right, considering this was just one AOF file being broken, I suspect you could restore all the other DB files except for this one and you'd get a 98%-intact database back. Glad to hear it was enough to get things working back though!  

The command it mentions could've also been ran from a WSL terminal hooked to the lanraragi distro to try and repair that broken file.

**Comment by Putarku** [comment]
When I was replacing the new version of the image, I encountered the same issue. So, I mapped the corrupted AOF file into another Redis container, ran the command "redis-check-aof --fix /data/appendonly.aof.14.incr.aof" to repair it, and then replaced the file back to its original location to start the container normally.

**Comment by denissslyu** [comment]
It happened to me 2 weeks ago, and i fixed that by redis-check-aof. And today, my rdb file CRC failed.
